sklearn/cluster/tests/test_hierarchical.py

```diff
@@ -140,6 +141,30 @@ def test_agglomerative_clustering_wrong_arg_memory():
     assert_raises(ValueError, clustering.fit, X)
 
 
+class Dummy(object):
+    def __init__(self):
+        pass
+
+    def cache(self, func):
+        return func
+
+
+class Wrong_Dummy(object):
+    def __init__(self):
+        pass
+
+
+def test_agglomerative_clustering_with_cache_attribute():
+    rng = np.random.RandomState(0)
+    n_samples = 100
+    X = rng.randn(n_samples, 50)
+    clustering = AgglomerativeClustering(memory=Dummy())
+    clustering.fit(X)
+
+    clustering = AgglomerativeClustering(memory=Wrong_Dummy())
+    assert_raises(ValueError, clustering.fit, X)
```

Comment:
Use assert_raises_message or assert_raises_regex to check the message which is raised.